config/initializers/vault.rb

```diff
@@ -114,5 +116,20 @@ def vault_hosts
     def vault_config_file
       ENV['VAULT_CONFIG_FILE'] || Rails.root.join("config/vault.json")
     end
+
+    # get back the vault instance that's required for this pod.  we'll take
+    # the key, and look up the instance or return all of 'em if it's a global key
+    def vault_instances(key = nil)
```

Comment:
I'd love to see some test that says a write goes to all vault instances when the deploy group is global / goes to 1 when it's local ...